internal/planner/planner.go

```diff
@@ -248,6 +248,7 @@ func (p *Planner) planRules(rules []*ast.Rule) (string, error) {
 				err := p.planQuery(rule.Body, 0, func() error {
 
 					// Run planner on the result.
+					p.loc = rule.Head.Loc() // reset location
```

Comment:
This is somewhat minor, but I wonder if we can fix `planQuery` so that when this inner function is called, the location set above (at L224) is intact. I think it would be best if callers can assume the location is not overwritten.

Reply:
:+1: done